matlab/test/arrow/type/tDateTypeTest.m

```diff
@@ -0,0 +1,80 @@
+% Shared superclass for DateType-related tests for Date32Type and Date64Type
+classdef tDateType < matlab.unittest.TestCase
```

Comment:
The original `tDate32Type` and `tDate64Type` test classes inherited from `hFixedWidthType`. I would suggest having `tDateType` inherit from `hFixedWidthType`. That way, you get the test cases defined in hFixedWidthType class for "free". 

Additionally, I would suggest renaming this class to `hDateTypeTest.m` because its a helper test class.